README.md

```diff
@@ -7,15 +7,198 @@
+harder to properly test all the possible states that a component can be in,
+since a store might contain state that isn't directly consumed by the component
+to be tested.
+
+But more importantly, we now need to manage a store during testing. This means
+we need to make sure we **always** restore it to its previous state in an
+`beforeEach`, otherwise this can lead to hard to debug failed tests.
+
+Instead of adding a `beforeEach`, we can also mock it out, which eliminates the
+need to reset the store.
+
+3. Reusing the component is possible
+
+While we focused implementing our own set of stores, some people prefer to use
+Redux, Rx, mobx or some other library for managing state and implementing
+unidirectional data flow.
+
+By storing state in an external store, we implicitly couple the component to
+whatever architecture we chose for our main application. If we're implementing
+an accordion component using Flux, it means everyone using our component will
+have to use Flux in order to interact with it (even though it might be hidden
+though the public API of the component).
+
+Hence using component state (and props) instead of stores is the preferred way
+when creating reusable components.
+
+## Connecting components using lifecycle methods
+
+So far we oftentimes "connected" components to stores by adding an event
+listener on the store in the component's lifecycle methods:
+
+```js
+class MyCounter extends React.Component {
+	componentDidMount() {
+		this.removeListener = store.addListener(count => {
+			this.setState({ count });
+		});
+	}
+	componentWillUnmount() {
+		this.removeListener();
+	}
+}
+```
+
+In `componentDidMount` as add an event listener that updates the component's
+state based on the stores state, thus triggering a re-rendering.
+
+In `componentWillUnmount` we remove the event listener. Typically `addListener`
+returns a function that — when invoked — removes the event listener from the
+store. In other cases people might subclass the `EventEmitter` class directly
+in order to implement a store, hence there are cases in which we actually have
+to call `removeListener` on the store itself.
+
+`componentDidMount` and `componentWillUnmount` are so-called lifecycle methods,
+meaning they are being run as part of the component being mounted (or
+unmounted).
+
+There are some other lifecycle methods, e.g. `componentWillMount` and
+`componentWillUpdate`. One thing to remember is that there is **no**
+`componentDidUnmount` method. This is because once a component is unmounted in
+"no longer exists" in a sense, hence doing any form of cleanup logic in there
+would be "too late".
+
+Whether or not we use `componentDidMount` instead of `componentWillMount`
+doesn't make a difference here. We can call `setState` in both.
+
+## Presentational vs Container Components
+
+When using a Flux or Flux-like architecture, it's generally speaking considered
+best-practice to avoid the number of container components.
```

Comment:
Not sure what's meant by 'avoid the number' here — avoiding too many container components?